ocaml/xapi/xapi_vdi.ml

```diff
@@ -25,6 +25,30 @@ open Printf
 (**************************************************************************************)
 (* current/allowed operations checking                                                *)
 
+let check_sm_feature_error op sm_features sr =
```

Comment:
@gaborigloi wrote
> This will also enable warnings/errors in case the match is incomplete, provided we enable the relevant compiler flags.

We have this even without the type annotation: to check on this, I commented out the line
```
  | `scan
```
and tried compiling, which gave an error at the `op` argument where the function is called:
```
File "ocaml/xapi/xapi_vdi.ml", line 210, characters 61-63:
Error: This expression has type API.vdi_operations
       but an expression was expected of type
         [< `blocked
          | `clone
          | `copy
          | `destroy
          | `force_unlock
          | `forget
          | `generate_config
          | `mirror
          | `resize
          | `resize_online
          | `snapshot
          | `update
          > `blocked `clone `copy `destroy `mirror `resize `resize_online
            `snapshot ]
       The second variant type does not allow tag(s) `scan
```
@gaborigloi also wrote,

> this type annotation will also check for typos in the variant names.

This works without the type annotation too: I tried it with
```
  | `scannnnnn
```
and got the same error as above.

Without the type annotation we do not detect an _additional_ nonsense variant-name, though: the code below builds without an error.
```
  | `scan
  | `extra_nonexistent_bogus_op
  | `force_unlock
```